source/Keyboard.cpp

```diff
@@ -1,35 +1,74 @@
+#include <unistd.h>
+#include <iostream>
+
 #include "Keyboard.hpp"
 #include "Server.hpp"
 
+
+std::map<std::string, uint32_t> modifiersLst = {
+  {"Alt", WLR_MODIFIER_ALT},
+  {"Ctrl", WLR_MODIFIER_CTRL}
+};
+
 Keyboard::Keyboard(Server *server, struct wlr_input_device *device) : server(server), device(device)
 {
+  shortcuts["Ctrl+Alt+t"] = {"Terminal", [](){
+    if (fork() == 0)
+    {
+      execl("/bin/sh", "/bin/sh", "-c", "weston-terminal", nullptr);
+    }
+  }};
+
+  shortcuts["a+F4"] = {"destroy", [server](){
+    View *view;
+    wl_list_for_each(view, &server->views, link) {
+      if (view->xdg_surface->role == WLR_XDG_SURFACE_V6_ROLE_TOPLEVEL &&
+          view->xdg_surface->toplevel->server_pending.activated)
+        view->close();
+    };
+  }};
 
+  shortcuts["Alt+Escape"] = {"Leave", [server](){ wl_display_terminate(server->display);}};
+  shortcuts["Ctrl+Escape"] = {"Leave", [server](){ wl_display_terminate(server->display);}};
 }
 
-bool Keyboard::handle_keybinding(xkb_keysym_t sym)
+Keyboard::~Keyboard() {
+  if (keymap) {
+		xkb_keymap_unref(keymap);
+	}
+  wl_list_remove(&key.link);
+	wl_list_remove(&modifiers.link);
+}
+
+bool Keyboard::handle_keybinding()
 {
-  switch (sym)
-    {
-    case XKB_KEY_Escape:
-      wl_display_terminate(server->display);
-      break;
-    case XKB_KEY_F1:
-      {
-	if (wl_list_length(&server->views) < 2)
-	  {
-	    break;
-	  }
-	View *current_view = wl_container_of(server->views.next, current_view, link);
-	View *next_view = wl_container_of(current_view->link.next, next_view, link);
-	ServerView::focus_view(next_view, next_view->xdg_surface->surface);
-	wl_list_remove(&current_view->link);
-	wl_list_insert(server->views.prev, &current_view->link);
-	break;
+  for (auto const & shortcut : shortcuts) {
+    char *tmp;
+    char *key = new char[shortcut.first.size() + 1];
+    shortcut.first.copy(key, shortcut.first.size() + 1);
+  	key[shortcut.first.size()] = 0;
+
+    uint32_t sum = 0;
+    uint32_t mod = 0;
+    tmp = strtok(key, "+");
+    while (tmp != NULL) {
+      if (modifiersLst.find(tmp) != modifiersLst.end()) {
+        mod += modifiersLst[tmp];
       }
-    default:
-      return false;
+      else
+        sum += xkb_keysym_from_name(tmp, XKB_KEYSYM_CASE_INSENSITIVE);
+      tmp = strtok(NULL, "+");
+    }
+    std::cout << "Shortcuts Code: " << sum << " + " << mod << " -> " << shortcut.second.name << std::endl;
+    if ((keycodes_states.last_raw_modifiers == mod) && sum == keycodes_states.sum) {
+      std::cout << keycodes_states.sum << " + " << keycodes_states.last_raw_modifiers << std::endl;
+      shortcut.second.action();
+      return true;
     }
-  return true;
+    delete key;
```

Comment:
Case in point for the comment about memory usage: `key` can be leaked since this delete may not be reached due to the return above.